core-connector/src/main/java/com/modusbox/client/processor/CustomErrorProcessor.java

```diff
@@ -23,44 +24,73 @@
                 try {
                     if (null != e.getResponseBody()) {
-                        /* Below if block needs to be changed as per the error object structure specific to 
+                        /* Below if block needs to be changed as per the error object structure specific to
                             CBS back end API that is being integrated in Core Connector. */
 
                         customJsonMessage.logJsonMessage("error", String.valueOf(exchange.getIn().getHeader("X-CorrelationId")),
                                 "Logging entire error object...", null, null,
                                 e.getResponseBody());
 
                         JSONObject respObject = new JSONObject(e.getResponseBody());
-                        if (respObject.has("message")) {
+                        if (respObject.has("message") && respObject.has("statusCode")) {
 //                            statusCode = String.valueOf(respObject.getInt("returnCode"));
 //                            errorDescription = respObject.getString("returnStatus");
                             statusCode = String.valueOf(respObject.getInt("statusCode"));
 //                            statusCode = respObject.getString("statusCode");\
                             // Replace 2 or more whitespace chars with just one
                             detailedDescription = respObject.getString("message").replaceAll("\\s+", " ");
                             try {
-                                errorMessage = respObject.getJSONObject("transferState").getJSONObject("lastError").getJSONObject("mojaloopError").getJSONObject("errorInformation").getString("errorDescription");
+                                originErrorMsg = respObject.getJSONObject("transferState").getJSONObject("lastError").getJSONObject("mojaloopError").getJSONObject("errorInformation").getString("errorDescription");
+
+                                jsonObjectMessage  = ErrorCode.getMojaloopErrorResponseByStatusCode(statusCode, locale);
+                                errorResponse      = new JSONObject(jsonObjectMessage).getJSONObject("errorInformation");
+                                errorMessage       = errorResponse.getString("description");
+                                errorMessageLocale = errorResponse.getString("descriptionLocale");
+
+                                //Update Rounding Value Error Message
+                                if(statusCode.equals("5241")){
+                                    String lastWord = originErrorMsg.substring(originErrorMsg.lastIndexOf(" ")+1);
+                                    System.out.println("Lastword of  Rounding Error String: "  + lastWord);
+                                    lastWord = CharMatcher.is('.').trimTrailingFrom(lastWord);
+
+                                    if (lastWord.length() > 0 && lastWord.matches("[0-9]+")) {
+                                        String numberRefined = lastWord.replaceAll("[^0-9]", "");
```

Comment:
@shunleiwaiko I think no need do this line because of already checked with regex.